### PR TITLE
Add tgenv makefile release target

### PR DESCRIPTION
**Pull request by austinguo550**
## Why? 🤔

This will make it much easier to create a release.

## What? :hammer_and_wrench:

Adding a Makefile target to create a git tag and release.

Makefile output:
```
> RELEASE=v0.2.1 make release
Creating release for v0.2.1
fatal: tag 'v0.2.1' already exists
Tag v0.2.1 already exists, creating release for tag
a CHANGELOG.md
a LICENSE
a Makefile
a README.md
a assets
a assets/tgenv-logo.png
a bin
a bin/terragrunt
a bin/tgenv
a libexec
a libexec/tgenv---version
a libexec/tgenv-list-remote
a libexec/tgenv-use
a libexec/tgenv-version-file
a libexec/tgenv-uninstall
a libexec/tgenv-upgrade
a libexec/tgenv-install
a libexec/tgenv-list
a libexec/tgenv-exec
a libexec/tgenv-version-name
a libexec/helpers
a libexec/tgenv-help
a list_all_versions_offline
a test
a test/helpers.sh
a test/setup_runner.sh
a test/run.sh
a test/test_symlink.sh
a test/install_deps.sh
a test/test_uninstall.sh
a test/test_list.sh
a test/test_install_and_use.sh
51dadf7b7965a67dd51376ad4fa08611ab98970c3af77c515ca7a018c8b08485  ./dist/tgenv-affirm.tar.gz
a release with the same tag name already exists: v0.2.1
Release v0.2.1 already exists, doing nothing
Please update brew formula: url and sha256
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ifndef RELEASE
+	$(error "RELEASE env var is undefined")
+endif
+ifndef PKG_NAME
+	$(error "PKG_NAME makefile var is undefined")
+endif
+	@echo "Creating release for $$RELEASE"
+	@git tag "$$RELEASE" || >&2 echo "Tag $$RELEASE already exists, creating release for tag"
+	@mkdir -p ./dist
+	@tar -czvf "./dist/$(PKG_NAME).tar.gz" .* --exclude ./dist
```
Comment (duganth) [inline Makefile]: had success with `@tar -czvf --exclude="dist" --exclude-vcs * "./dist/$(PKG_NAME).tar.gz"`

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,21 @@
+.PHONY: release
+
+PKG_NAME := "tgenv-affirm"
+
+release:
+	@which gh >/dev/null || >&2 echo "Please install gh CLI with \"brew install gh\""
```
Comment (duganth) [inline Makefile]: This do not error out of make just FYI

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+endif
+ifndef PKG_NAME
+	$(error "PKG_NAME makefile var is undefined")
+endif
+	@echo "Creating release for $$RELEASE"
+	@git tag "$$RELEASE" || >&2 echo "Tag $$RELEASE already exists, creating release for tag"
+	@mkdir -p ./dist
+	@tar -czvf "./dist/$(PKG_NAME).tar.gz" .* --exclude ./dist
+	@sha256sum "./dist/$(PKG_NAME).tar.gz"
+	@gh release create --repo Affirm/tgenv "$$RELEASE" --verify-tag "./dist/$(PKG_NAME).tar.gz"
```
Comment (duganth) [inline Makefile]: Did you want to push tags in this script as well?

File: Makefile (hunk 1 above)
Comment (austinguo550) [reply Makefile]: yup, the problem was the `.*`. swapped it with just `*`

File: Makefile (hunk 3 above)
Comment (austinguo550) [reply Makefile]: yes we do, I just added that

File: Makefile (hunk 2 above)
Comment (austinguo550) [reply Makefile]: yup fixed that now

**Review by duganth — APPROVED** [review]

**Pull request merged**